test/test_model.py

```diff
@@ -91,6 +92,64 @@ def test_average_model():
     assert np.all(tmodel_2000.predict([0, 0, 0]) == 1100)
 
 
+def test_gp_model(datadir):
+    # ToDo
+    # What if we are in the multi-shell case ?
+    # Assume single shell case for now
+    num_gradients = 1
+    # a = np.zeros(num_gradients)  # acquisition parameters
+    # h = nib.load()  # Susceptibility induced off‐resonance field (Hz)
+    # ToDo
+    # Provide proper values/estimates for these
+    a = 1
+    h = 1  # should be a NIfTI image
+
+    # ToDo
+    # Build the kernel properly following the paper.
+    # Also, needs to be a sklearn.gaussian_process.kernels.Kernel instance:
+    # https://scikit-learn.org/stable/modules/generated/sklearn.gaussian_process.kernels.Kernel.html
+    # kernel = np.ones(20)
+    kernel = DotProduct() + WhiteKernel()
+
+    dwi = DWI.from_filename(datadir / "dwi.h5")
+
+    _dwi_data = dwi.dataobj
+    # Use a subset of the data for now to see that something is written to the
+    # output
+    # bvecs = dwi.gradients[:3, :].T
+    bvecs = dwi.gradients[:3, 10:13].T  # b0 values have already been masked
+    # bvals = dwi.gradients[3:, 10:13].T  # Only for inspection purposes: [[1005.], [1000.], [ 995.]]
+    dwi_data = _dwi_data[60:63, 60:64, 40:45, 10:13]
+
+    num_iterations = 5
+    gp = model.GaussianProcessModel(
+        dwi=dwi, a=a, h=h, kernel=kernel, num_iterations=num_iterations
+    )
+    indices = list(range(bvecs.shape[0]))
+    # ToDo
+    # This should be done within the GP model class
+    # Apply lovo strategy properly
+    # Vectorize and parallelize
+    result = np.zeros_like(dwi_data)
+    for idx in indices:
+        lovo_idx = np.ones(len(indices), dtype=bool)
+        lovo_idx[idx] = False
+        X = bvecs[lovo_idx]
+        for i in range(dwi_data.shape[0]):
+            for j in range(dwi_data.shape[1]):
+                for k in range(dwi_data.shape[2]):
+                    # ToDo
+                    # Use a mask to avoid traversing background data
+                    y = dwi_data[i, j, k, lovo_idx]
+                    gp.fit(X, y)
+                    prediction, _ = gp.predict(
+                        bvecs[idx, :][np.newaxis]
+                    )  # Can take multiple values X[:2, :]
+                    result[i, j, k, idx] = prediction.item()
+
+    assert result.shape == dwi_data.shape
```

Comment:
Let's use the notebook for this, for the time being. The only tests to add at this point are:
- The model can be instatiated
- We can fit and predict one voxel - data could be drawn from scikit-learn's examples so that one of the default kernels is used.

```suggestion
```

Reply:
Done in 061f130.